extensions/spring-xd-extension-reactor/src/main/java/org/springframework/xd/integration/reactor/net/ReactorPeerFactoryBean.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright 2013-2014 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.xd.integration.reactor.net;
+
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+import reactor.Environment;
+import reactor.core.Dispatcher;
+import reactor.io.buffer.Buffer;
+import reactor.io.codec.Codec;
+import reactor.io.codec.DelimitedCodec;
+import reactor.io.codec.LengthFieldCodec;
+import reactor.io.codec.StandardCodecs;
+import reactor.io.net.ChannelStream;
+import reactor.io.net.NetStreams;
+import reactor.io.net.ReactorPeer;
+import reactor.io.net.Spec;
+import reactor.io.net.codec.syslog.SyslogCodec;
+import reactor.io.net.impl.netty.http.NettyHttpServer;
+import reactor.io.net.impl.netty.tcp.NettyTcpServer;
+import reactor.io.net.impl.netty.udp.NettyDatagramServer;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * @author Stephane Maldini
+ */
+@SuppressWarnings({"unchecked", "rawtypes"})
+public class ReactorPeerFactoryBean<IN, OUT> implements FactoryBean<ReactorPeer<IN, OUT, ?>> {
```

Comment:
This is a pretty big new addition - what does this do at a high level, can you add javadoc?